staging/src/k8s.io/apiserver/plugin/pkg/kmsprovider/vault/client.go

```diff
@@ -0,0 +1,212 @@
+/*
```

Comment:
All the code for interacting with vault should be contained to `k8s.io/kubernetes/plugin/pkg/auth/kmsprovider/vault`.  It must not be inside of k8s.io/apiserver

Reply:
Are you suggesting to split the code in two packages 1 that goes in k8s.io/apiserver/plugin/pkg (as your pervious comment) and 1 that goes in k8s.io/kubernetes/plugin/pkg/auth?  Please elaborate.

Reply:
> Are you suggesting to split the code in two packages 1 that goes in k8s.io/apiserver/plugin/pkg (as your pervious comment) and 1 that goes in k8s.io/kubernetes/plugin/pkg/auth? Please elaborate.

Ah, came up on the apimachinery call.  Daniel's preference it put the dependency in k8s.io/kubernetes/plugin, not k8s.io/apiserver/plugin .   I don't feel strongly either way, it'll live in k8s.io/kubernetes

Reply:
done.